### ISSUE TITLE
JSON error? Invalid control character at line 1: column ...

### ISSUE THREAD

**Issue by ghostduck** [edwardslabs/cleverwrap.py#12]
Hi there, I recent just encountered a strange error when calling the `say()` function.
`CleverWrap(bot.cleverbot_key).say(msg)`

However it gaves us an error related to JSON. The backtrace is like:
```
/cleverwrap/cleverwrap.py, line 60, in say
reply = self._send(params)

/cleverwrap/cleverwrap.py, line 77, in_send
return r.json()

...(skipped)
builtins.ValueError: Invalid control character at: line 1 column number (char that number-1)
```

The skipped part if you are interested:
```
python3/dist-packages/requests/models.py, line 793 in json
return json.loads(self.text, **kwargs)

python3.4/json/__init__.py, line 318, in loads
return _default_decoder.decode(s)

python3.4/json/decoder.py
obj, end = self.raw_decode(s, idx=w(s, 0).end())

python3.4/json/decoder.py, line 359, in raw_decode
obj, end = self.scan_once(s, idx)
```

We don't see that error after restarting our app. 
Also, unfortunately we can't see the JSON response when that error occurred.
Any ideas about this error?

**Comment by edwardslabs** [comment]
I have hit this a few times on my application as well. I think it has to do with a bad conversation state and conversation ID. If you call CleverWrap(bot.cleverbot_key).reset() it should fix the issue. I am still trying to figure out why it gets in this state though.

**Comment by ghostduck** [comment]
More findings, hope the information here will be helpful.

Our chatbot uses Cleverwrap to say something in chatroom.

When that error occur, the JSON has a length close to, or not less than 2000. 
`builtins.ValueError: Invalid control character at: line 1 column 1933 (char that 1932)`

We checked the cleverbot API page after creating this issue. 
Then, we notice the API will record conversation state(cs) and cs will keep growing. Our chat members talk with the bot crazily and that error occurs.

Is it possible that there are upper limit related to cs or other variables?

**Comment by edillingham** [comment]
Is there a specific set of steps that can be used to reproduce this problem?  I was not able to trigger the exception even after the size of the "cs" value was > 3000 bytes.

**Comment by ghostduck** [comment]
Sorry, I don't really know the way to reproduce the problem.

We just keep calling `cw.say(msg)` in our bot when our users say something to the bot.

`builtins.ValueError: Invalid control character at: line 1 column 1113 (char 1112)` The error seems can occur as short as 1100 characters.

**Comment by linuxdaemon** [comment]
The issue appears to be characters below `0x20` appearing in the data, which is considered invalid by the JSON spec. 

After rigorous testing I tracked the issue (in my case at least) to some random characters in one of the `interaction` fields in the API reply. 

This can be replicated by just doing `json.loads('"\\x03"')` since `0x03` is not an allowed character in a JSON string. The fix seems to be `json.loads('"\\x03"', strict=False)` which disables the control code restrictions.

Fixed in #15
